app/src/main/java/se/rebeccazadig/bokholken/login/UserRepository.kt

```diff
@@ -1,17 +1,25 @@
 package se.rebeccazadig.bokholken.login
 
+import android.util.Log
+import com.google.firebase.auth.FirebaseUser
 import com.google.firebase.auth.ktx.auth
 import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.delay
 
 class UserRepository private constructor() {
 
     private val myAuth = Firebase.auth
 
+    suspend fun saveUser(user: User) {
+        delay(2000)
+        Log.i("Emma", "Saving user ${user.name}, ${user.contact}, ${user.city}")
```

Comment:
You don't print id. Is it empty or not?

Reply:
suspend fun saveUser(user: User): Result {
        delay(2_000)
        return withContext(Dispatchers.IO) {
            if (user.name.startsWith("A", ignoreCase = true)) {
                Log.i("Emma", "Saving user ${user.name}, ${user.contact}, ${user.city}, ${user.id}")
                Result.Success
            } else {
                val errorMessage = "Name must start with 'A'"
                Result.Failure(errorMessage)
            }
        }
    }

I am printing user.id. Don't know why you haven't got the correct changes. I don't have any changes to commit